common/pkg/testutils/capitest/patches.go

```diff
@@ -39,55 +39,57 @@ type JSONPatchMatcher struct {
 }
 
 func ValidateGeneratePatches[T mutation.GeneratePatches](
-	t *testing.T,
+	t GinkgoTInterface,
 	handlerCreator func() T,
 	testDefs ...PatchTestDef,
 ) {
 	t.Helper()
+	testFunc := func(tt *PatchTestDef) {
+		g := gomega.NewWithT(t)
+		h := handlerCreator()
+		req := &runtimehooksv1.GeneratePatchesRequest{
+			Variables: tt.Vars,
+			Items: []runtimehooksv1.GeneratePatchesRequestItem{
+				tt.RequestItem,
+				{
+					HolderReference: runtimehooksv1.HolderReference{
+						APIVersion: capiv1.GroupVersion.String(),
+						Kind:       "Cluster",
+						Namespace:  request.Namespace,
+						Name:       request.ClusterName,
+					},
+				},
+			},
+		}
+		resp := &runtimehooksv1.GeneratePatchesResponse{}
+		h.GeneratePatches(context.Background(), req, resp)
+		expectedStatus := runtimehooksv1.ResponseStatusSuccess
+		if tt.ExpectedFailure {
+			expectedStatus = runtimehooksv1.ResponseStatusFailure
+		}
+		g.Expect(resp.Status).
+			To(gomega.Equal(expectedStatus), fmt.Sprintf("Message: %s", resp.Message))
+
+		if len(tt.ExpectedPatchMatchers) == 0 {
+			g.Expect(resp.Items).To(gomega.BeEmpty())
+			return
+		}
+		g.Expect(resp.Items).To(containPatches(&tt.RequestItem, tt.ExpectedPatchMatchers...))
 
+		if len(tt.UnexpectedPatchMatchers) > 0 {
+			g.Expect(resp.Items).
+				ToNot(containPatches(&tt.RequestItem, tt.UnexpectedPatchMatchers...))
+		}
+	}
+
+	// compose Ginkgo table arguments
+	// https://onsi.github.io/ginkgo/#table-specs for more details
+	testArgs := make([]TableEntry, 0, len(testDefs))
 	for testIdx := range testDefs {
 		tt := testDefs[testIdx]
-
-		t.Run(tt.Name, func(t *testing.T) {
-			t.Parallel()
```

Comment:
The region and httpproxy have few more test cases. Ginkgo runs them in parallel. Even github runners shows same time as they take on `main` branch. however the the number of testcases reported are very different. 
I am not sure why # of test cases decreased. It should report increase in number of tests. 🤔 
This PR:
```
DONE 154 tests in 138.937s
```
Other PR
```
DONE 239 tests in 137.412s
```
checking it out.

Reply:
The decrease in number of test cases are because Ginkgo's `DescribeTable` spec will consider a set of test cases as single case. The actual test coverage increased.
 Region: `pkg/handlers/aws/mutation/region (1.09s) (coverage: 16.7% of statements)1` VS `pkg/handlers/aws/mutation/region (1.095s) (coverage: 80.0% of statements)` 

HTTPProxy: `pkg/handlers/generic/mutation/httpproxy (1.107s) (coverage: 50.0% of statements)` VS `pkg/handlers/generic/mutation/httpproxy (7.915s) (coverage: 77.2% of statements)`

I am not sure how to configure ginkgo to count the test cases same as go tests does. 
But we are good here. All test cases are executed as expected.